src/modules/roc_fec/reader.cpp

```diff
@@ -288,16 +312,18 @@ void Reader::update_source_packets_() {
         n_fetched++;
 
         if (packet::blknum_lt(fec->source_block_number, cur_sbn_)) {
-            roc_log(LogTrace, "fec reader: dropping source packet from previous block:"
-                              " cur_sbn=%lu pkt_sbn=%lu pkt_sn=%lu",
-                              (unsigned long)cur_sbn_,
-                              (unsigned long)fec->source_block_number,
-                              (unsigned long)rtp->seqnum);
+            roc_log(LogTrace,
+                    "fec reader: dropping source packet from previous block:"
+                    " cur_sbn=%lu pkt_sbn=%lu pkt_sn=%lu",
+                    (unsigned long)cur_sbn_, (unsigned long)fec->source_block_number,
+                    (unsigned long)rtp->seqnum);
             n_dropped++;
             continue;
         }
 
         roc_panic_if(fec->source_block_number != cur_sbn_);
+        update_size_(fec->source_block_length);
```

Comment:
As discussed, we should do this only at the beginning of a block.

Reply:
(see #180)

Reply:
Discussed to set `alive_ = false` in the current PR.